src/components/side_panel/components/collapse/collapse.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, onMounted, onWillUpdateProps, useRef } from "@odoo/owl";
+import { SpreadsheetChildEnv } from "../../../../types";
+
+interface Props {
+  open: boolean;
+  slots: any;
+}
+
+export class Collapse extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-Collapse";
+  static props = {
+    open: Boolean,
+    slots: Object,
+  };
+
+  private contentRef = useRef("content");
+
+  setup() {
+    onMounted(() => {
+      this.setCollapsedState(this.props.open, true);
+    });
+
+    onWillUpdateProps((nextProps) => {
+      if (nextProps.open !== this.props.open) {
+        this.setCollapsedState(nextProps.open);
+      }
+    });
+  }
+
+  setCollapsedState(open: boolean, skipTransition = false) {
+    const el = this.contentRef.el;
+    if (!el) return;
+
+    if (open) {
+      el.style.maxHeight = el.scrollHeight + "px";
```

Comment:
doesn't work at 100% :x Open the chart design panels and click on the button to add a trendline. `el.scrollHeight` is only valid at the moment of the transition, but this isn't updated when the content size is updated afterwards so it will overflow.

We probably could fix that by removing the max-height property on `transitionend` event ? Or a simple setTimeout.

Reply:
We should maybe keep bootstrap approach. A `d-none` when it's collapsed, nothing when it's not collapsed, when starting a collapse manually set the size to 0/scrollHeight, add a class/style with `transition`, set the target height, and at the end of the transition remove everything (I think that's what they do)